src/test/java/de/ait/secondlife/security/controllers/AuthIntegrationTest.java

```diff
@@ -0,0 +1,210 @@
+                                "}"))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void return_400_for_non_existent_user_email() throws Exception {
+        mockMvc.perform(post("/v1/auth/user/login")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\n" +
+                                "  \"email\": \"usertest@test.com\",\n" +
+                                "  \"password\": \"qwerty-123\"\n" +
+                                "}"))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void return_401_for_incorrect_user_password() throws Exception {
+        mockMvc.perform(post("/v1/auth/user/login")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\n" +
+                                "  \"email\": \"test.user@test.com\",\n" +
+                                "  \"password\": \"123\"\n" +
+                                "}"))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    public void return_new_access_token_for_user() throws Exception {
+        MvcResult loginResult = mockMvc.perform(post("/v1/auth/user/login")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\n" +
+                                "  \"email\": \"test.user@test.com\",\n" +
+                                "  \"password\": \"qwerty-123\"\n" +
+                                "}"))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        MockHttpServletResponse loginResponse = loginResult.getResponse();
+        String loginContent = loginResponse.getContentAsString();
+        String refreshToken = JsonPath.parse(loginContent).read("$.refreshToken");
+
+        MvcResult accessResult = mockMvc.perform(post("/v1/auth/user/access")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\n" +
+                                "  \"refreshToken\": \"" + refreshToken + "\"\n" +
+                                "}"))
+                .andExpect(status().isOk())
+                .andReturn();
+        MockHttpServletResponse accessResponse = accessResult.getResponse();
+        String accessContent = accessResponse.getContentAsString();
+        assertNotNull(JsonPath.parse(accessContent).read("$.accessToken"));
+        assertNotNull(JsonPath.parse(accessContent).read("$.refreshToken"));
+        assertNotNull(accessResponse.getCookie("Access-Token").getValue());
+    }
+
+    @Test
+    public void return_400_for_refreshing_with_invalid_user_refresh_token() throws Exception {
+        mockMvc.perform(post("/v1/auth/user/access")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\n" +
+                                "  \"refreshToken\": \"invalidUserRefreshToken\"\n" +
```

Comment:
i think this test should verify valid refresh token and invalid and return result of comparison. For the particular test need create a user and set him some refreshToken, and send invalid token and compare result

Reply:
not necessary